pex/environment.py

```diff
@@ -214,18 +218,24 @@ def mount(
         target=None,  # type: Optional[Target]
     ):
         # type: (...) -> PEXEnvironment
+        pex_file = os.path.realpath(pex)
         if not pex_info:
-            pex_info = PexInfo.from_pex(pex)
+            pex_info = PexInfo.from_pex(pex_file)
             pex_info.update(PexInfo.from_env())
         pex_hash = pex_info.pex_hash
         if pex_hash is None:
             raise AssertionError(
                 "There was no pex_hash stored in {} for {}.".format(PexInfo.PATH, pex)
             )
-        pex_root = pex_info.pex_root
-        pex = maybe_install(pex=pex, pex_root=pex_root, pex_hash=pex_hash) or pex
         target = target or targets.current()
-        return cls(pex=pex, pex_info=pex_info, target=target)
+        key = (pex_file, pex_hash, target)
+        mounted = cls._CACHE.get(key)
```

Comment:
This new `mount` caching combines with the pre-existing instance-level `resolve()` caching to avoid resolving twice in the happy path of the interpreter that tries to run a PEX 1st being a compatible one.